### PR TITLE
remove all usages of future.standard_library

### PR DESCRIPTION
**Pull request by megies**
It can be problematic due to it's monkey patching, see #1599.

#1613 might have to be adjusted..

### CODE REVIEW

**Comment by krischer** [comment]
This actually makes #1613 quite a bit easier as we do not have to migrate all clients to `requests` anymore. I'll rebase #1613 once this one here is merged.

**Comment by megies** [comment]
Locally run docker tests across **all** modules look promising, but some things need to be looked at: http://tests.obspy.org/?node=docker-&git=af684575e06bf069

File: obspy/db/scripts/indexer.py [hunk 1]
```diff
@@ -28,17 +28,14 @@
                         unicode_literals)
 from future.builtins import *  # NOQA @UnusedWildImport
 from future.utils import native_str
-from future import standard_library
 
+import http.server
```
Comment (QuLogic) [inline obspy/db/scripts/indexer.py]: This uses the direct import from `future` that was only added in 0.14.0.

Comment (megies) [inline obspy/db/scripts/indexer.py]: Oh.. I thought I was importing directly in my clean system-Python but somehow future was doing a man-in-the-middle attack on me there.. ;-)

**Comment by megies** [comment]
All full docker tests look OK, unrelated fails. http://tests.obspy.org/?node=docker-&git=b87c2a2

This one has a lot of arclink issues: http://tests.obspy.org/62532/

There are some lxml issues i can't remember from before, e.g.: http://tests.obspy.org/62532/#1

Maybe somebody can double-check but I think we can merge this.

**Comment by krischer** [comment]
Puh...this are honestly a lot of failures and I have a hard time explaining many of them. Are they some temporary struggles on IRIS' side? I was under the impression that the event service will stay online at least for a while longer.

**Comment by megies** [comment]
> this are honestly a lot of failures and I have a hard time explaining many of them

Try running fdsn tests on 1.0.2 or a recent maintenance_1.0.x, at least some of them are in there too, so likely a recent server side change.

Others are related to #1500, there should be conditional skips for those...

I agree there's a chance there might be some new ones, but better get rid of the unrelated ones to see clearer first.

**Comment by krischer** [comment]
I just restarted the travis builds. Lets see what happens.

**Comment by megies** [comment]
Rebased and force pushed (after merge of #1618) to see a bit clearer through the test fails..

**Comment by QuLogic** [comment]
It seems that there are still some client errors, and somehow it's failing with minimum SciPy dep, but I don't see how that might have changed here.

**Comment by Jollyfant** [comment]
It seems that there are still some client errors, and somehow it's failing with minimum SciPy dep, but I don't see how that might have changed here.

Perhaps because the service changed and is now returning nothing (204). So when trying to parse that response it fails.

**Comment by megies** [comment]
Those syngine fails are also in current maintenance, so they can be disregarded. The minimum SciPy fails for fdsn mass downloader are know failures (see #1500), so they can be disregarded too.

Comparing the Travis run of current maintenance and this PR, I think we can safely merge this one, it seems to me like there are some erratic fails that are happening unrelatedly.

 - arclink fails seem totally erratic, before the rebase we had lots of arclink fails in this PR, now we have lots of arclink fails on maintenance and here only a few
 - some IRIS FDSNWS helpstring fails that also seem erratic, we have one fail on this PR and one fail (in a different build) on maintenance

Travis comparison..

|  | [maintenance_1.0.x](https://travis-ci.org/obspy/obspy/builds/183052393) | [this PR](https://travis-ci.org/obspy/obspy/builds/183052588) |
| --- | --- | --- |
| Linux Py2.7 (min. dep.)  | https://travis-ci.org/obspy/obspy/jobs/183052394#L919  | https://travis-ci.org/obspy/obspy/jobs/183052589#L929  |
| Linux Py2.7  | https://travis-ci.org/obspy/obspy/jobs/183052397#L957  | https://travis-ci.org/obspy/obspy/jobs/183052590#L968  |
| Linux Py3.3  | https://travis-ci.org/obspy/obspy/jobs/183052398#L748  | https://travis-ci.org/obspy/obspy/jobs/183052591#L763  |
| Linux Py3.4  | https://travis-ci.org/obspy/obspy/jobs/183052399#L775  | https://travis-ci.org/obspy/obspy/jobs/183052592#L796  |
| Linux Py3.5  | https://travis-ci.org/obspy/obspy/jobs/183052400#L771  | https://travis-ci.org/obspy/obspy/jobs/183052593#L791  |
| MacOs Py2.7  | https://travis-ci.org/obspy/obspy/jobs/183052401#L839  | https://travis-ci.org/obspy/obspy/jobs/183052594#L858  |
| MacOs Py3.5  | https://travis-ci.org/obspy/obspy/jobs/183052402#L673  | https://travis-ci.org/obspy/obspy/jobs/183052595#L694  |

**Comment by megies** [comment]
Rebased on current maintenance_1.0.x + force pushed to get new test results.

**Comment by megies** [comment]
CI is all green for this PR now, only exception is known and unrelated failures of FDSN mass downloader on minimum scipy version.

So, should be OK to merge this, I think.

**Comment by megies** [comment]
A reason not to merge this, anybody? If not I will merge this.

**Comment by megies** [comment]
Merging..

**Pull request merged**